sqlite3_vtable_test.go

```diff
@@ -148,8 +151,335 @@ func TestCreateModule(t *testing.T) {
 			t.Fatalf("want %v but %v", intarray[i], value)
 		}
 	}
+
 	_, err = db.Exec("DROP TABLE vtab")
 	if err != nil {
 		t.Fatalf("couldn't drop virtual table: %v", err)
 	}
 }
+
+func TestVUpdate(t *testing.T) {
+	tempFilename := TempFilename(t)
+	defer os.Remove(tempFilename)
+
+	// create module
+	updateMod := &vtabUpdateModule{t, make(map[string]*vtabUpdateTable)}
+
+	// register module
+	sql.Register("sqlite3_TestVUpdate", &SQLiteDriver{
+		ConnectHook: func(conn *SQLiteConn) error {
+			return conn.CreateModule("updatetest", updateMod)
+		},
+	})
+
+	// connect
+	db, err := sql.Open("sqlite3_TestVUpdate", tempFilename)
+	if err != nil {
+		t.Fatalf("could not open db: %v", err)
+	}
+
+	// create test table
+	_, err = db.Exec(`CREATE VIRTUAL TABLE vt USING updatetest(f1 integer, f2 text, f3 text)`)
+	if err != nil {
+		t.Fatalf("could not create updatetest vtable vt, got: %v", err)
+	}
+
+	// check that table is defined properly
+	if len(updateMod.tables) != 1 {
+		t.Fatalf("expected exactly 1 table to exist, got: %d", len(updateMod.tables))
+	}
+	if _, ok := updateMod.tables["vt"]; !ok {
+		t.Fatalf("expected table `vt` to exist in tables")
+	}
+
+	// check nothing in updatetest
+	rows, err := db.Query(`select * from vt`)
+	if err != nil {
+		t.Fatalf("could not query vt, got: %v", err)
+	}
+	i, err := getRowCount(rows)
+	if err != nil {
+		t.Fatalf("expected no error, got: %v", err)
+	}
+	if i != 0 {
+		t.Fatalf("expected no rows in vt, got: %d", i)
+	}
+
+	_, err = db.Exec(`delete from vt where f1 = 'yes'`)
```

Comment:
line 220 -- the delete is there to check that a delete still works when the table is empty. It doesn't have much to do with the actual vUpdate stuff, but I get carried away when writing unit tests to make sure start conditions are always the same.